src/buildstream/_artifact.py

```diff
@@ -52,7 +52,7 @@
 #
 class Artifact:
 
-    version = 1
+    version = 2
```

Comment:
I had assumed that this value would match the version in src/buildstream/_protos/buildstream/v2/artifact.proto which clams to only need bumping when mandatory fields change.. are these in fact unrelated? or should the other be bumped also?

Reply:
The line:

```
  int32 version = 1;
```

Indicates that the *field number* in the artifact proto is 1, this number 1 will never change.

The indicates that it need not be incremented for an optional field, however given that buildstream requires this field *instead* of the buildtree for some operations, it's probably more prudent to just bump the field.

In any case, since we bump the `BST_ARTIFACT_VERSION` we will anyway be loading different artifacts.